app/models/course/assessment/question/programming.rb

```diff
@@ -110,22 +110,37 @@ def non_autograded_template_files=(template_files)
 
   private
 
+  # Create new package or re-evaluate the old package.
+  def process_package
+    if attachment_changed?
+      attachment ? process_new_package : remove_old_package
+    elsif time_limit_changed? || memory_limit_changed? || language_id_changed?
```

Comment:
That's a good question 😄 

Now it will raise an `NotImplementedError` if we change the root language in UI ( I did some tests just now).  

If that part get implemented or fixed, I think it will go the first branch (`attachment_changed?`) if the root language version changed, because root language change means the deletion of old package and new package is generated.